lib/engine/part/city.rb

```diff
@@ -53,11 +53,18 @@ def tokenable?(corporation, free: false, tokens: nil)
           next false unless get_slot(t.corporation)
           next false if !free && t.price > corporation.cash
           next false if @tile.cities.any? { |c| c.tokened_by?(t.corporation) }
+          next true if @reservations.index(corporation.name)
```

Comment:
why do you need this one?

Reply:
I originally had

next false if !@reservations.index(corporation.name) && @tile.reservations.count { |x| corporation.name != x } >= @tile.cities.sum(&:available_slots)

but rubocop said it was too long :)

The alternative is to see if @tile.reservations.count is >=1 and then do the check on the available slots.

It gets complicated in multicity case otherwise.